raiden/network/pathfinding.py

```diff
@@ -264,3 +302,51 @@ def query_paths(
             'Pathfinding service returned invalid json',
             dict(response_text=response.text, exc_info=True),
         )
+
+
+def query_paths(
+        service_config: typing.Dict[str, typing.Any],
+        our_address: typing.Address,
+        privkey: bytes,
+        current_block_number: typing.BlockNumber,
+        token_network_address: typing.Union[typing.TokenNetworkAddress, typing.TokenNetworkID],
+        route_from: typing.InitiatorAddress,
+        route_to: typing.TargetAddress,
+        value: typing.TokenAmount,
+) -> typing.List[typing.Dict[str, typing.Any]]:
+
+    max_paths = service_config['pathfinding_max_paths']
+    url = service_config['pathfinding_service_address']
+    payload = {'from': route_from, 'to': route_to, 'value': value, 'max_paths': max_paths}
+    offered_fee = service_config.get('pathfinding_fee', service_config['pathfinding_max_fee'])
+    scrap_existing_iou = False
+
+    for retries in reversed(range(MAX_PATHS_QUERY_ATTEMPTS)):
```

Comment:
Instead of reversing the range, you could do `range(MAX_PATHS_QUERY_ATTEMPTS, -1, -1)`

I can see that `MAX_PATHS_QUERY_ATTEMPTS` is set to 2 which means that my range would run 3 times (the second param is telling range where to stop at -1... so the range would be [2, 1, 0], you can set it to 0 if you want to just try 2 times)

Reply:
I thought `reverse(range(MAX_PATHS_QUERY_ATTEMPTS))` looks less awkward than `range(MAX_PATHS_QUERY_ATTEMPTS - 1, -1, -1)` - i want the loop to run 2 times and `retries` to be 0 in the last iteration.